Apply attribute value limit for BYTESLICE and KindBytes (#7990)
open-telemetry/opentelemetry-go: CHANGELOG.md

### Added

- Add `ByteSlice` and `ByteSliceValue` functions for new `BYTESLICE` attribute type in `go.opentelemetry.io/otel/attribute`. (#7948)

> - Apply attribute value limit to the `BYTESLICE` attribute type in `go.opentelemetry.io/otel/sdk/trace`. (#7990)

Reviewer: We need to also to do it for `go.opentelemetry.io/otel/auto`.
See https://github.com/open-telemetry/opentelemetry-go/pull/8153/files#r3065034756

> - Apply attribute value limit to the `KindBytes` attribute type in `go.opentelemetry.io/otel/trace`. (#7990)

Reviewer: ```suggestion
- Apply attribute value limit to the `BYTESLICE` attribute type in `go.opentelemetry.io/otel/sdk/trace`. (#7990)
- Apply attribute value limit to the `BYTESLICE` attribute type in `go.opentelemetry.io/otel/trace`. (#7990)
```

> - Apply attribute value limit to the `KindBytes` attribute type in `go.opentelemetry.io/otel/sdk/log`. (#7990)

Reviewer: ```suggestion
- Apply attribute value limit to the `KindBytes` attribute type in `go.opentelemetry.io/otel/sdk/log`. (#7990)
- Apply attribute value limit to the `KindBytes` attribute type in `go.opentelemetry.io/otel/trace`. (#7990)
```

Reviewer: Why have you resolved this? Any reason why do not want to mention `go.opentelemetry.io/otel/trace`? @MrAlias, or maybe it should be `go.opentelemetry.io/auto`?

```suggestion
- Apply attribute value limit to the `KindBytes` attribute type in `go.opentelemetry.io/otel/sdk/log`. (#7990)
- Apply attribute value limit to the `KindBytes` attribute type in `go.opentelemetry.io/otel/auto`. (#7990)
```

Reviewer 2: I think leaving that out makes sense. Not a big issue for the Go auto devs to find it.
